content/movie-review-pt2.md

```diff
@@ -0,0 +1,522 @@
+
+Once the system program creates a new account for us, we can now populate the data field of the account using the format of a struct type we specified in `state.rs`. We start by reference the `pda_account` (that we that we just created) and borrowing the data field on the account. This allows us to access the data without taking ownership. The `try_from_sliced_unchecked` function deserializes the data field of the borrowed account, using the format of our `MovieAccountState` struct.
+
+We’ll assign this data to a variable called `account_data`. We now have access to the data field of the `pda_account` and are ready to continue with our checks and updates. We first check if the account has already been initialized by using the `is_initilized` function implemented for our `MovieAccountState` struct in `state.rs`. If the account has not been initialized, then we populate the additional fields (rating, title description) on the `MovieAccountState` struct using the inputs provided by the user. Lastly, we set the initialized field to true.
+
+```rust
+	msg!("unpacking state account");
+	let mut account_data = try_from_slice_unchecked::<MovieAccountState>(&pda_account.data.borrow()).unwrap();
+  msg!("borrowed account data");
+
+  msg!("checking if movie account is already initialized");
+  if account_data.is_initialized() {
+      msg!("Account already initialized");
+      return Err(ProgramError::AccountAlreadyInitialized);
+  }
+
+  account_data.title = title;
+  account_data.rating = rating;
+  account_data.description = description;
+  account_data.is_initialized = true;
+```
+
+## Serialize Account Data
+
+Our `account_data` variable now holds the deserialized data in the format of our `MovieAccountState` struct updated with the user’s inputs for the movie review. We now need to save the data on the `account_data` variable to the data field on our actual `pda_account`. We do this by borrowing a mutable reference to the data field of the `pda_account` again and then serializing (convert to bytes) the `account_data` to the data field on the `pda_account`. This effectively saves the updated `account_data` variable to the data field of the new account. Now when a user fetches the `pda_account` and deserializes (convert from bytes) the data field, it will display the updated data we’ve serialized into the account.
+
+```rust
+account_data.serialize(&mut &mut pda_account.data.borrow_mut()[..])?;
+```
+
+# Demo
+
+Let’s practice this together by continuing to work on the Movie Review program from the last lesson. No worries if you’re just jumping into this lesson - it should be possible to follow either way.
+
+As a refresher, we are building a Solana program which lets users review movies. Last lesson, we deserialized the instruction data passed in by the user but we have not yet store this data in an account. Let’s now update our program to create new accounts to store the user’s movie review.
+
+****Download the starter code****
```

Comment:
@ZYJLiu could you turn these into numbered H3 headeers?

Reply:
fixed